include/hyper.deal/operators/advection/advection_operation.h

```diff
@@ -303,31 +308,61 @@ namespace hyperdeal
             dealii::AlignedVector<VNumber> scratch_data_array;
             scratch_data_array.resize_fast(dealii::Utilities::pow(n_points, dim) * dim_x);
             VNumber *tempp = scratch_data_array.begin();
+            
+            if(do_skew)
+              {
+                if (dim_x >= 1) eval_.template gradients<0, true, false>(buffer, tempp + dealii::Utilities::pow(n_points, dim) * 0);
+                if (dim_x >= 2) eval_.template gradients<1, true, false>(buffer, tempp + dealii::Utilities::pow(n_points, dim) * 1);
+                if (dim_x >= 3) eval_.template gradients<2, true, false>(buffer, tempp + dealii::Utilities::pow(n_points, dim) * 2);
+              }
+            
             for (auto qv = 0u, q = 0u; qv < dealii::Utilities::pow<unsigned int>(n_points, dim_v); ++qv)
               for (auto qx = 0u; qx < dealii::Utilities::pow<unsigned int>(n_points, dim_x); ++qx, ++q)
                 {
                   VNumber    grad_in[dim_x];
                   const auto vel = velocity_field->evaluate_x(q, qx, qv);
+                  
+                  if(do_skew)
+                    phi.template submit_value<false>(data_ptr, - factor_skew * (phi.get_gradient_x(tempp, q, qx, qv) * vel), q, qx, qv);
+                  
                   for (int d = 0; d < dim_x; d++)
-                    grad_in[d] = buffer[q] * vel[d];
+                    grad_in[d] = (1.0-factor_skew) * buffer[q] * vel[d];
                   phi.submit_gradient_x(tempp, grad_in, q, qx, qv);
                 }
-            if (dim_x >= 1) eval_.template gradients<0, false, false>(tempp + dealii::Utilities::pow(n_points, dim) * 0, data_ptr);
+            
```

Comment:
The part starting from here and up to line 339 should be incapsulated into `if (factor_skew != 1.0)`